server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -294,6 +294,10 @@ private ClusterFormationState getClusterFormationState() {
         );
     }
 
+    public Optional<String> getClusterFormationWarning() {
+        return clusterFormationFailureHelper.emittedWarning();
```

Comment:
I started out making tweaks to this PR for this and wound up just putting up this draft instead -- https://github.com/elastic/elasticsearch/pull/87306. Before I put more effort into that, do you think it's appropriate to expose all of ClusterFormationState? We seem to need most of the information in it (if not all of it), but it does seem a little odd to publicly expose a private record like that.

Reply:
It's only a private record because we've never done anything with this info except log it before. Definitely makes sense to me to make it public and use it more widely.